src/sentry/notifications/utils/participants.py

```diff
@@ -262,6 +262,17 @@ def disabled_users_from_project(project: Project) -> Mapping[ExternalProviders,
     return output
 
 
+def filter_users_by_project_membership(users: List[RpcUser], project: Project) -> List[RpcUser]:
+    """
+    Returns a list of users that belong to the project
+
+    `users`: The list of users
+    `project`: The project that users' project membership will be checked for
+    """
+    project_members = project.member_set.values()
```

Comment:
i think this would be really expensive to load all the members for the project

is it possible to filter get_many_by_email by project id https://github.com/getsentry/sentry/blob/master/src/sentry/services/hybrid_cloud/user/impl.py#L33-L40

i see something else in there about project_ids but i wasn't sure how this worked https://github.com/getsentry/sentry/blob/master/src/sentry/services/hybrid_cloud/user/impl.py#L85-L90

Reply:
thank you, this is a great idea!